benchmark/python/dot.py

```diff
@@ -83,198 +208,236 @@ def get_iter(path, data_shape, batch_size):
-
-        csr_data = []
-        dns_data = []
-        num_batch = 0
-        for batch in train_iter:
-            data = train_iter.getdata()
-            csr_data.append(data)
-            dns_data.append(data.tostype('default'))
-            num_batch += 1
-        bag_of_data = [csr_data, dns_data]
-        num_repeat = 5
-        costs = []
-        for d in bag_of_data:
-            weight.wait_to_read()
-            cost = 0.
-            count = 0
-            for d_batch in d:
-                d_batch.wait_to_read()
-                cost += measure_cost(True, num_repeat, mx.nd.dot, d_batch, weight)
-                count += 1
-            costs.append(cost/count)
-        t_sparse = costs[0]
-        t_dense = costs[1]
-        ratio = t_dense / t_sparse
-        print('density(%)\tn\tm\tk\tt_dense/t_sparse\tt_dense\tt_sparse')
-        fmt = "%0.4f\t\t%d\t%d\t%d\t%0.2f\t\t\t%0.4f\t%0.6f"
-        print(fmt % (density * 100, batch_size, m, k, ratio, t_dense, t_sparse))
-
-
-def test_dot_synthetic():
+def test_dot_synthetic(data_dict):
     """benchmark sparse mxnet dot and scipy dot operator with matrices of given density.
-    `t_sparse` is the runtime of the invoked sparse dot operator in ms, while `t_dense` is the 
+    `t_sparse` is the runtime of the invoked sparse dot operator in ms, while `t_dense` is the
     runtime of dot(dns, dns), with the same matrices except that they are in default storage type.
     """
-    # Benchmark MXNet's sparse dot operator
-    def bench_mx_dot(lhs_shape, rhs_shape, lhs_stype, rhs_stype, lhs_den, rhs_den, trans_lhs, ctx, repeat):
+    # Benchmark MXNet and Scipys dot operator
+    def bench_dot(lhs_shape, rhs_shape, lhs_stype, rhs_stype,
+                  lhs_den, rhs_den, trans_lhs, ctx, num_repeat=10, fw="mxnet", distribution="uniform"):
         set_default_context(ctx)
+        assert fw == "mxnet" or fw == "scipy"
+        # Set funcs
+        dot_func_sparse = mx.nd.dot if fw == "mxnet" else sp.spmatrix.dot
+        dot_func_dense = mx.nd.dot if fw == "mxnet" else np.dot
         # Create matrix instances
-        lhs_nd = rand_ndarray(lhs_shape, lhs_stype, density=lhs_den)
-        rhs_nd = rand_ndarray(rhs_shape, rhs_stype, density=rhs_den)
-        lhs_dns = lhs_nd if lhs_stype == 'default' else lhs_nd.tostype('default')
-        rhs_dns = rhs_nd if rhs_stype == 'default' else rhs_nd.tostype('default')
-        # One warm up run, verify correctness
-        out = mx.nd.dot(lhs_nd, rhs_dns, trans_lhs)
-        out_expected = mx.nd.dot(lhs_dns, rhs_dns, trans_lhs)
-        assert_almost_equal(out.asnumpy(), out_expected.asnumpy(), rtol=1e-2, atol=1e-3)
-        # Start benchmarking
+        lhs_nd = rand_ndarray(lhs_shape, lhs_stype, density=lhs_den, distribution=distribution)
+        # only uniform distribution supported for rhs
+        rhs_nd = rand_ndarray(rhs_shape, rhs_stype, density=rhs_den, distribution="uniform")
         lhs_nd.wait_to_read()
         rhs_nd.wait_to_read()
```

Comment:
since you have `mx.nd.waitall()` in `measure_cost`, these wait_to_read() can be removed